src/main/kotlin/com/demonwav/mcdev/platform/mcp/at/psi/AtMixinUtil.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+@file:JvmName("AtMixinUtil")
+package com.demonwav.mcdev.platform.mcp.at.psi
+
+import com.demonwav.mcdev.util.findQualifiedClass
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiClass
+import com.intellij.psi.impl.JavaPsiFacadeEx
+import com.intellij.psi.search.GlobalSearchScope
+
+fun getClassFromString(text: String?, project: Project): PsiClass? {
+    var newText = text ?: return null
+
+    // We don't care about arrays
+    newText = newText.replace("\\[".toRegex(), "")
+
+    val scope = GlobalSearchScope.allScope(project)
+
+    if (newText.startsWith("L")) {
+        // class
+        newText = newText.substring(1)
+
+        newText = newText.replace(";", "")
+        newText = newText.replace("/", ".")
+
+        return findQualifiedClass(project, newText, scope)
+    }
+
+    val instance = JavaPsiFacadeEx.getInstanceEx(project)
+
+    instance.apply {
+        // Primitive
+        when (newText) {
+            "B" -> return findClass(Int::class.java.canonicalName, scope)
```

Comment:
You could use `PsiBytecodeUtil.getPrimitiveType(newText).getBoxedType` here for some reduced code duplication